plexpy/notifiers.py

```diff
@@ -2721,3 +2748,94 @@ def return_config_options(self):
+        self.incl_subject = plexpy.CONFIG.HIPCHAT_INCL_SUBJECT
+        self.emoticon = plexpy.CONFIG.HIPCHAT_EMOTICON
+
+    def notify(self, message, subject):
+        if not message or not subject:
+            return
+        
+        if self.incl_subject:
+            text = subject.encode('utf-8') + ': ' + message.encode('utf-8')
+        else:
+            text = message.encode('utf-8')
+
+        if self.emoticon:
+            text = self.emoticon + ' ' + text
+
+        data = {'color': self.color,
+                'message': text,
+                'notify': 'false',
+                'message_format': 'text'}
+
+        hiphost = urlparse(self.apiurl).hostname
+        hippath = urlparse(self.apiurl).path
+        hipquery = urlparse(self.apiurl).query
+        hipfullq = hippath + '?' + hipquery
+
+        http_handler = HTTPSConnection(hiphost)
+        http_handler.request("POST",
+                             "%s" % (hipfullq),
+                             headers={'Content-type': "application/json"},
+                             body=json.dumps(data))
+        response = http_handler.getresponse()
+        request_status = response.status
+
+        if request_status == 200 or request_status == 204:
+            logger.info(u"PlexPy Notifiers :: Hipchat notification sent.")
+            return True
+        elif request_status >= 400 and request_status < 500:
+            logger.warn(u"PlexPy Notifiers :: Hipchat notification failed: [%s] %s" % (request_status, response.reason))
+            return False
+        else:
+            logger.warn(u"PlexPy Notifiers :: Hipchat notification failed.")
+            return False
+
+    def test(self, apiurl, color, hipchat_emoticon, hipchat_incl_subject):
+
+        self.enabled = True
+        self.apiurl = apiurl
+        self.color = color
+        self.emoticon = hipchat_emoticon
+        self.incl_subject = hipchat_incl_subject
+
+        return self.notify('PlexPy', 'Test Message')
+
+    def return_config_options(self):
+        config_option = [{'label': 'Hipchat Custom Integrations Full URL',
+                          'value': self.apiurl,
+                          'name': 'hipchat_url',
+                          'description': 'Your Hipchat integration URL. You can get a key from'
+                                         ' <a href="' + helpers.anon_url('https://www.hipchat.com/addons/') + '" target="_blank">here</a>.',
+                          'input_type': 'text'
+                          },
+                         {'label': 'Hipchat Color',
+                          'value': self.color,
+                          'name': 'hipchat_color',
+                          'description': 'Background color for the message.',
+                          'input_type': 'select',
+                          'select_options': {'yellow': 'yellow', 'green': 'green', 'red': 'red', 'purple': 'purple', 'gray': 'gray', 'random': 'random'}
```

Comment:
The api will use 'gray' if it is not specified.  Would it be more appropriate if I set that as the default in config.py?